sdk/identity/identity/src/credentials/deviceCodeCredential.ts

```diff
@@ -78,33 +76,34 @@ export class DeviceCodeCredential implements TokenCredential {
     userPromptCallback: DeviceCodePromptCallback = defaultDeviceCodePromptCallback,
     options?: TokenCredentialOptions
   ) {
-    this.tenantId = tenantId;
-    this.clientId = clientId;
+    if (!tenantId.match(/^[0-9a-zA-Z-.:/]+$/)) {
```

Comment:
It'd be nice if this regex check and error reporting could be moved to a util method somewhere since it's showing up in multiple files